WPMedia/fusion-news-theme-blocks
# TMEDIA-169 - Lazy load list blocks

## Description

Add Lazy Load functionality to the follow list blocks

* Card List
* Numbered List
* Results List
* Simple List
* Search Results

Also made a small adjustment to the top table list block to match


## Jira Ticket
- [TMEDA-169](https://arcpublishing.atlassian.net/browse/TMEDIA-169)

## Acceptance Criteria
1. A new boolean custom field called Lazy load block? should be added to all blocks listed in the section above
    * If Lazy load block? is true, the block should lazy load on the page once it comes within 300px of the viewport
    * If Lazy load block? is false, the block should load immediately (it should not lazy load) – this is the existing behavior
2. The default value for this custom setting should be false
3. Server sider render of the feature will not return and HTML
4. Client side should fetch content if needed (Verify using network panel to see API call)
5. LazyLoad should not be active within the PageBuilder Editor UI

## Test Steps

1. Checkout this branch `git checkout TMEDIA-169-lazy-load-list-blocks`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/card-list-block,@wpmedia/numbered-list-block,@wpmedia/results-list-block,@wpmedia/simple-list-block,@wpmedia/search-results-list-block,@wpmedia/top-table-list-block`
3. Either on a new page or existing page toggle on the new custom field option "lazyLoad" each of the blocks.
4. View the published page
5. Verify the HTML for each feature is not present on page load - You can disable JavaScript in dev tools and verify the features you enabled lazyLoad for do not render on server render
6. Verify when you scroll the page that as you begin to get nearer the features there are network calls to fetch the content needed for each feature you have enabled lazyLoad for.


Search uses a template - would need to update feature on template to enable lazy load custom field
http://localhost/pf/search/story/?_website=the-gazette

Simple List
Homepage - right rail
http://localhost/homepage/?_website=the-gazette

Results List
Used on section fronts, update Section template - 
http://localhost/news/?_website=the-gazette

Numbered List
Homepage - right rail
http://localhost/homepage/?_website=the-gazette

Card List
Homepage - main
http://localhost/homepage/?_website=the-gazette

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._
